internal/target.go

```diff
@@ -44,34 +44,13 @@ func (t Target) Scan() (sbom.SBOM, error) {
 	if err != nil {
 		return sbom.SBOM{}, fmt.Errorf("failed to create source from %q: %w", t.Path, err)
 	}
-	result := sbom.SBOM{
-		Source: src.Describe(),
-		Descriptor: sbom.Descriptor{
-			Name:    "syft",
-			Version: version.SyftVersion,
-		},
-	}
-
-	// Enable all the image catalogers to mimic same cataloging behavior as syft running on an image
-	config := cataloger.DefaultConfig()
-	catalogers := cataloger.ImageCatalogers(config)
-	config.Catalogers = make([]string, len(catalogers))
-	for i, c := range catalogers {
-		config.Catalogers[i] = c.Name()
-	}
-
-	packageCatalog, relationships, theDistro, err := syft.CatalogPackages(src, config)
-	if err != nil {
-		return sbom.SBOM{}, err
-	}
-
-	result.Artifacts.Packages = packageCatalog
-	result.Artifacts.LinuxDistribution = theDistro
-	result.Relationships = relationships
 
+	result, err := syft.CreateSBOM(context.Background(), src, syft.DefaultCreateSBOMConfig())
```

Comment:
It looks like earlier the image catalogers were being used regardless of the input source type. To replicate the same behavior you can add that ask to the `CreateSBOMConfig`:
```golang

cfg := syft.DefaultCreateSBOMConfig().
		WithCatalogerSelection(
			pkgcataloging.NewSelectionRequest().
				WithDefaults(pkgcataloging.ImageTag),
		)
```